src/internal/components/pluggableVisualizations/pivotTable/widthItemsHelpers.ts

```diff
@@ -0,0 +1,168 @@
+    IBucketFilter,
+    isAttributeFilter,
+    IBucketItem,
+} from "../../../interfaces/Visualization";
+
+const isAttributeColumnWidthItemVisible = (
+    _widthItem: IAttributeColumnWidthItem,
+    _filters: IBucketFilter[],
+): boolean => true;
+
+const isMeasureWidthItemMatchedByFilter = (
+    widthItem: IMeasureColumnWidthItem,
+    filter: IAttributeFilter,
+): boolean =>
+    filter.selectedElements.some(selectedElement =>
+        widthItem.measureColumnWidthItem.locators.some(
+            locator =>
+                !isMeasureLocatorItem(locator) &&
+                locator.attributeLocatorItem.element === selectedElement.uri,
+        ),
+    );
+
+const isMeasureColumnWidthItemVisible = (
+    widthItem: IMeasureColumnWidthItem,
+    filters: IBucketFilter[],
+): boolean =>
+    filters.reduce((isVisible, filter) => {
+        if (isAttributeFilter(filter)) {
+            const shouldBeMatched = !filter.isInverted;
+            return isVisible && shouldBeMatched === isMeasureWidthItemMatchedByFilter(widthItem, filter);
+        }
+        return isVisible;
+    }, true);
+
+const isWidthItemVisible = (widthItem: ColumnWidthItem, filters: IBucketFilter[]): boolean =>
+    isAttributeColumnWidthItem(widthItem)
+        ? isAttributeColumnWidthItemVisible(widthItem, filters)
+        : isMeasureColumnWidthItemVisible(widthItem, filters);
+
+const containsMeasureLocator = (widthItem: IMeasureColumnWidthItem): boolean =>
+    widthItem.measureColumnWidthItem.locators.some(locator => isMeasureLocatorItem(locator));
+
+const widthItemLocatorsHaveProperLength = (
+    widthItem: IMeasureColumnWidthItem,
+    measuresCount: number,
+    columnAttributesCount: number,
+): boolean => {
+    return (
+        widthItem.measureColumnWidthItem.locators.length > 0 &&
+        ((measuresCount > 0 &&
+            containsMeasureLocator(widthItem) &&
+            widthItem.measureColumnWidthItem.locators.length === columnAttributesCount + 1) ||
+            (measuresCount === 0 &&
+                !containsMeasureLocator(widthItem) &&
+                widthItem.measureColumnWidthItem.locators.length === columnAttributesCount))
+    );
+};
+
+// removes attribute widthItems with invalid identifiers
+// removes measure widthItems with invalid identifiers and invalid number of locators
+function adaptWidthItemsToPivotTable(
+    originalColumnWidths: ColumnWidthItem[],
+    measureLocalIdentifiers: string[],
+    rowAttributeLocalIdentifiers: string[],
+    columnAttributeLocalIdentifiers: string[],
+    filters: IBucketFilter[],
+): ColumnWidthItem[] {
+    const attributeLocalIdentifiers = [...rowAttributeLocalIdentifiers, ...columnAttributeLocalIdentifiers];
+
+    return originalColumnWidths.reduce((columnWidths: ColumnWidthItem[], columnWidth: ColumnWidthItem) => {
+        if (isMeasureColumnWidthItem(columnWidth)) {
+            // filter out invalid locators
+            const filteredMeasureColumnWidthItem: IMeasureColumnWidthItem = {
```

Comment:
Again, I would consider separating to removeInvalidLocators() so that you could have method name instead of the comment.

Reply:
Done